pkg/discovery/metrics/metric.go

```diff
@@ -153,39 +154,51 @@ type Metric interface {
 	GetMetricProp() MetricProp
 	GetUID() string
 	GetValue() interface{}
+	UpdateValue(existing interface{})
 }
 
 type MetricFilterFunc func(m Metric) bool
 
 type ResourceMetric struct {
 	resourceType ResourceType
 	metricProp   MetricProp
-	value        float64
+	value        interface{}
+}
+
+type Points struct {
+	Values []float64
+	// Time at which the last metric value was appended
+	Timestamp int64
 }
 
-func NewResourceMetric(rType ResourceType, mProp MetricProp, v float64) ResourceMetric {
+func NewResourceMetric(rType ResourceType, mProp MetricProp, v interface{}) ResourceMetric {
 	return ResourceMetric{
 		resourceType: rType,
 		metricProp:   mProp,
 		value:        v,
 	}
 }
 
-type EntityResourceMetric struct {
+type EntityDetails struct {
 	metricUID string
 
 	entityType DiscoveredEntityType
 	entityID   string
+}
 
+type EntityResourceMetric struct {
+	EntityDetails
 	ResourceMetric
 }
 
 func NewEntityResourceMetric(eType DiscoveredEntityType, id string, rType ResourceType, mProp MetricProp,
 	v float64) EntityResourceMetric {
```

Comment:
It would be helpful to have this input `v` as `interface{}` instead of `float64`

Reply:
Fixed!